packages/graph/hash_graph/lib/graph/src/api/rest/middleware.rs

```diff
@@ -196,7 +215,36 @@ pub fn span_maker(request: &hyper::Request<hyper::Body>) -> tracing::Span {
         http.client_ip = %client_ip,
         http.user_agent = %user_agent,
         http.host = %host,
+        http.status_code = Empty,
+        otel.status_code = Empty,
     );
     tracing_opentelemetry::OpenTelemetrySpanExt::set_parent(&span, remote_context);
     span
 }
+
+fn span_on_response(
+    response: &hyper::Response<axum::body::BoxBody>,
+    _latency: Duration,
+    span: &tracing::Span,
+) {
+    let status = response.status().as_u16();
+    span.record("http.status_code", &tracing::field::display(status));
+    span.record("otel.status_code", "OK");
```

Comment:
Is this always `"OK"` or should it change depending on `status`?

Reply:
Is `on_response` always equivalent to `on_success`? In my head failures would also be responses

Reply:
Have you checked if these both trigger, and if they do does the order matter? Will `on_response` override the value from `on_failure` or vice versa?

Reply:
From the spec, this should only be `"OK"` or `"ERROR"` https://opentelemetry.io/docs/reference/specification/common/mapping-to-non-otlp/#span-status

It does seem like we shouldn't actually set the `"OK"` state manually as it would override any error codes set by other parts of the application code. I think we can remove this `otel.status_code` record, but keep the one for the failure case

Reply:
Removed setting it in 50cad89ec9ed07786b7b71850a7fcf33ce231e4d